server/src/main/java/com/objectcomputing/checkins/services/skills/SkillServices.java

```diff
@@ -0,0 +1,76 @@
+package com.objectcomputing.checkins.services.skills;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+public class SkillServices {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SkillServices.class);
+
+    @Inject
+    private SkillRepository skillsRepo;
+
+    protected Skill saveSkill(Skill skill) {
+
+        List<Skill> returnedList = findByValue(null, skill.getName(), null);
+        return returnedList.size() < 1 ? skillsRepo.save(skill) : null;
+
+    }
+
+    protected Skill readSkill(UUID skillId) {
+
+        Skill returned = skillsRepo.findBySkillid(skillId);
+
+        return returned;
+
+    }
+
+    protected List<Skill> findByValue(UUID skillid, String name, Boolean pending) {
+        List<Skill> skillList = null;
+        if (skillid != null) {
+           skillList = Collections.singletonList(readSkill(skillid));
+        } else if(name != null) {
+            skillList = findByNameLike(name);
+        }  else if(pending != null) {
+            skillList = findByPending(pending);
+        }
```

Comment:
This is fine for now...however, I think there is a way to do a "findByILike" that takes an example entity and does case insensitive matching on *any and all* of the populated values.  I'd have to dig into that a little bit more...but as this stands, if I wanted to do a search for name="Java" and pending=true, I couldn't do that. it would only search by name and would return anything that matched regardless of pending.